### PR TITLE
Bin Realtime data

### PR DESCRIPTION
**Pull request by sok-leaphea**
#screenshot
image

### CODE REVIEW

File: components/BinList.js [hunk 1]
```diff
@@ -1,55 +1,116 @@
...
-            { text: 'Level', color: 'blue' },
-            { text: 'Location', color: 'black' }
-        ]
-    }
-];
+import { getDatabase, ref, onValue, off } from 'firebase/database'; 
+import { database } from '../firebaseConfig'; 
+import { useNavigation } from '@react-navigation/native'; // Import useNavigation
 
 const Bins = () => {
```
Comment (heayem) [inline components/BinList.js]: should name the same file name

File: components/BinList.js [hunk 2]
```diff
@@ -1,55 +1,116 @@
...
+            const bins = Object.keys(data).map(key => ({ id: key, ...data[key] }));
+            setBinData(bins);
+          } else {
+            setError('No data found');
+          }
+        }, {
+          onlyOnce: false, 
+        });
+      } catch (error) {
+        setError(error.message);
```
Comment (heayem) [inline components/BinList.js]: where message come? can user understand your message?

File: components/BinList.js [hunk 3]
```diff
@@ -1,55 +1,116 @@
...
+    return () => {
+      off(dbRef); 
+    };
+  }, []);
+
+  const handleLocationPress = (bin) => {
+    navigation.navigate('Map');
+  };
+
+  if (loading) {
```
Comment (heayem) [inline components/BinList.js:59]: should spinner

File: components/BinList.js [hunk 4]
```diff
@@ -1,55 +1,116 @@
...
+    return <Text style={styles.loadingText}>Loading...</Text>;
+  }
+  
+  if (error) {
+    return <Text style={styles.errorText}>Error: {error}</Text>;
```
Comment (heayem) [inline components/BinList.js]: message error  or success should be alert or taost

File: screens/BinData.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+      } finally {
+        setLoading(false);
+      }
+    };
+
+    fetchAllData();
+
+    const handleRealTimeUpdate = (update) => {
+      if (update.success) {
+        console.log('Real-time update:', update.data);
```
Comment (heayem) [inline screens/BinData.js]: remove console.log

**Review by heayem — APPROVED** [review]

File: components/BinList.js (hunk 1 above)
Comment (sok-leaphea) [reply components/BinList.js]: Noted

File: components/BinList.js [hunk 6]
```diff
@@ -27,7 +28,7 @@ const Bins = () => {
           onlyOnce: false, 
         });
       } catch (error) {
-        setError(error.message);
+        setError('Something went wrong!');
```
Comment (heayem) [inline components/BinList.js:31]: Hmm :(

**Review by heayem — APPROVED** [review]

**Review by heayem — APPROVED** [review]

**Review by heayem — APPROVED** [review]

**Pull request merged**